src/rdkafka.h

```diff
@@ -1018,6 +1018,26 @@ void rd_kafka_conf_set_events(rd_kafka_conf_t *conf, int events);
 
 
 /**
+ * @brief Creates/configures the SSL context using the given callback
+ *
+ * SSL contexts are usually created by providing the locations of the
+ * various files containing certificates and keys. With this set,
+ * the client has the ability to configure and set up the context
+ * to theirs needs.
+ *
+ * This is called once during initialization and can override any other
+ * ssl configuration if set (called after regular rdkafka ssl
+ * configuration).
+ *
+ */
+RD_EXPORT
+void rd_kafka_conf_set_ssl_ctx_cb (rd_kafka_conf_t *conf,
+			      rd_kafka_resp_err_t (*ssl_ctx_cb) (rd_kafka_t *rk,
+			                                         void *ssl_ctx,
```

Comment:
If a user compiles librdkafka and dynamically links it later, would this ifdef-guard work? The define might not be in the CU.